packages/core/src/types/oas3_1.ts

```diff
@@ -0,0 +1,586 @@
+    security: listOf('SecurityRequirement'),
+    servers: listOf('Server'),
+    requestBody: 'RequestBody',
+    responses: 'ResponsesMap',
+    deprecated: {
+      type: 'boolean',
+    },
+    callbacks: mapOf('Callback'),
+    'x-codeSamples': listOf('XCodeSample'),
+    'x-code-samples': listOf('XCodeSample'), // deprecated
+  },
+  required: ['responses'],
+};
+
+const XCodeSample: NodeType = {
+  properties: {
+    lang: { type: 'string' },
+    label: { type: 'string' },
+    source: { type: 'string' },
+  },
+};
+
+const RequestBody: NodeType = {
+  properties: {
+    description: {
+      type: 'string',
+    },
+    required: {
+      type: 'boolean',
+    },
+    content: 'MediaTypeMap',
+  },
+  required: ['content'],
+};
+
+const MediaTypeMap: NodeType = {
+  properties: {},
+  additionalProperties: 'MediaType',
+};
+
+const MediaType: NodeType = {
+  properties: {
+    schema: 'Schema',
+    examples: mapOf('Example'),
+    encoding: mapOf('Encoding'),
+  },
+};
+
+const Example: NodeType = {
+  properties: {
+    value: { isExample: true },
+    summary: {
+      type: 'string',
+    },
+    description: {
+      type: 'string',
+    },
+    externalValue: {
+      type: 'string',
+    },
+  },
+};
+
+const Encoding: NodeType = {
+  properties: {
+    contentType: {
+      type: 'string',
+    },
+    headers: mapOf('Header'),
+    style: {
+      enum: ['form', 'simple', 'label', 'matrix', 'spaceDelimited', 'pipeDelimited', 'deepObject'],
+    },
+    explode: {
+      type: 'boolean',
+    },
+    allowReserved: {
+      type: 'boolean',
+    },
+  },
+};
+
+const Header: NodeType = {
+  properties: {
+    description: {
+      type: 'string',
+    },
+    required: {
+      type: 'boolean',
+    },
+    deprecated: {
+      type: 'boolean',
+    },
+    allowEmptyValue: {
+      type: 'boolean',
+    },
+    style: {
+      enum: ['form', 'simple', 'label', 'matrix', 'spaceDelimited', 'pipeDelimited', 'deepObject'],
+    },
+    explode: {
+      type: 'boolean',
+    },
+    allowReserved: {
+      type: 'boolean',
+    },
+    schema: 'Schema',
+    examples: mapOf('Example'),
+    content: 'MediaTypeMap',
+  },
+};
+
+const ResponsesMap: NodeType = {
+  properties: {
+    default: 'Response',
+  },
+  additionalProperties: (_v: any, key: string) =>
+    responseCodeRegexp.test(key) ? 'Response' : undefined,
+};
+
+const Response: NodeType = {
+  properties: {
+    description: {
+      type: 'string',
+    },
+    headers: mapOf('Header'),
+    content: 'MediaTypeMap',
+    links: mapOf('Link'),
+  },
+  required: ['description'],
+};
+
+const Link: NodeType = {
+  properties: {
+    operationRef: { type: 'string' },
+    operationId: { type: 'string' },
+    parameters: null, // TODO: figure out how to describe/validate this
+    requestBody: null, // TODO: figure out how to describe/validate this
+    description: { type: 'string' },
+    server: 'Server',
+  },
+};
+
+const Schema: NodeType = {
```

Comment:
you should add more fields here, new json schema draft supports `if`, `else`, `const`, etc.